generic/ltp.py

```diff
@@ -130,6 +149,52 @@ def setUp(self):
         build.make(ltp_dir)
         build.make(ltp_dir, extra_args='install')
 
+        ltp_tstnet_dir = os.path.join(self.ltpbin_dir, 'testcases/bin/tst_net.sh')
+        if not os.path.exists(ltp_tstnet_dir):
+            self.log.info("File tst_net.sh does not exist")
+
+        # modifying input parameters in tst_net.sh file to run ltp on public address instead of localhost
+        filehandle = open(ltp_tstnet_dir, "r")
+        listOfLines = filehandle.readlines()
+        filehandle.close()
+
+        lineNo = [149, 150, 210, 211, 212, 988, 989, 990, 991, 1081, 1082, 1086, 1087]
+        newline = ["        LHOST_IFACES=\"${LHOST_IFACES:-" + self.host_iface + "}\"\n", "        RHOST_IFACES=\"${RHOST_IFACES:-" + self.peer_iface + "}\"\n", "        tst_require_cmds ssh\n", "        use=\"SSH\"\n", "        rcmd=\"ssh -q " + self.peer_user + "@" + self.peer_public_ip + "\"\n", "IPV4_LHOST=\"${IPV4_LHOST:-" + self.host_ipv4 + "/" + str(self.prefix) + "}\"\n", "IPV4_RHOST=\"${IPV4_RHOST:-" + self.peer_ipv4 + "/" + str(self.prefix) + "}\"\n", "IPV6_LHOST=\"${IPV6_LHOST:-" + self.host_ipv6 + "}\"\n", "IPV6_RHOST=\"${IPV6_RHOST:-" + self.peer_ipv6 + "}\"\n", "export LHOST_IFACES=\"${LHOST_IFACES:-" + self.host_iface + "}\"\n", "export RHOST_IFACES=\"${RHOST_IFACES:-" + self.peer_iface + "}\"\n", "export LHOST_HWADDRS=\"${LHOST_HWADDRS:-" + self.host_hwaddr + "}\"\n", "export RHOST_HWADDRS=\"${RHOST_HWADDRS:-" + self.peer_hwaddr + "}\"\n"]
+
+        for i, val in enumerate(lineNo):
+            listOfLines[val] = newline[i]
+            filehandle = open(ltp_tstnet_dir, "w")
+            filehandle.writelines(listOfLines)
+
+        filehandle.seek(0)
+        filehandle.truncate()
+        for i, line in enumerate(listOfLines):
+            if i not in [206, 207, 208, 209, 213]:
+                filehandle.write(line)
+
+        # necessary services to be started on the host for ltp cases successfull execution
+        services = ['nfs-server', 'rpcbind', 'httpd', 'vsftpd']
+        for service in services:
+            cmd = "service " + service + " restart"
+            if process.system(cmd, ignore_status=True, shell=True) != 0:
+                self.cancel("Unable to start the service")
+
+        # performing a silent ssh to remote machine to execute ltp commands on remote machine
+        send_cert = 'ssh-copy-id -f -i ' + '/root/.ssh/id_rsa.pub' + ' ' + self.peer_user + '@' + self.peer_public_ip
+        try:
+            child = pexpect.spawn(send_cert,encoding='utf-8')
+            child.sendline('yes')
+            child.expect('password:')
+            child.sendline(self.peer_password)
+            time.sleep(2)
```

Comment:
why not use  ssh.session()  instead of pexpect  to run commands over peer ?

Reply:
@abdhaleegit Here in this case we are not performing ssh from host machine to remote machine using avocado utils.
SSH is perform in tst_net.sh file for running some commands on remote machine and the ssh performed is without password. 
so i am creating a ssh key copy of remote machine and saving it in host machine, now whenever tst_net.sh script performs ssh without password, ssh happens successfully and commands will be executed on remote machine.

https://github.com/linux-test-project/ltp/blob/6246d1d067a505cf4889449cb83b4732111f8cc1/testcases/lib/tst_net.sh#L213